app/tasks/data_extractor_nlp.py

```
@@ -23,15 +42,150 @@ async def extract_and_process(self, request: ExtractDataModel) -> Dict[str, Any]
+
+    def _extract_text_content(self, soup: BeautifulSoup) -> str:
+        """텍스트 컨텐츠 추출"""
+        # HTML을 문자열로 변환하여 extract_main_text 함수 사용
+        html_content = str(soup)
+        return extract_main_text(html_content)
+
+    async def _extract_keywords_tfidf(self, text: str, max_keywords: int = 3) -> List[str]:
+        """TF-IDF를 사용한 키워드 추출"""
+        if not text:
+            return []
+        
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)